quail/src/main/java/com/mineinjava/quail/pathing/PathFollower.java

```diff
@@ -46,31 +54,42 @@ public void setPath(com.mineinjava.quail.odometry.path path) {
      * This does return a field-centric movement vector. It does not limit acceleration (yet)
      * @return the next movement to follow the path
      */
-    public robotMovement calculateNextDriveMovement() {
+    public RobotMovement calculateNextDriveMovement() {
         // calculate the next movement to follow the path
         Pose2d currentPose = this.localizer.getPoseEstimate();
-        double deltaAngle = util.deltaAngle(currentPose.heading, this.path.finalHeading);
+        double deltaAngle = Util.deltaAngle(currentPose.heading, this.path.getCurrentPoint().heading); // this may or may not work
+
         double turnSpeed = turnController.getOutput(0, deltaAngle);
         turnSpeed /= this.path.length();
```

Comment:
You need to change this to be turnspeed/distance to next point. This is a remnant from when rotation was done over the entire path and not between points.